config/presets/fastnet.go

```diff
@@ -92,7 +92,7 @@ func fastnet() config.Config {
 	conf.Beacon.VotesLimit = 100
 
 	conf.POET.GracePeriod = 10 * time.Second
-	conf.POET.CycleGap = 30 * time.Second
+	conf.POET.CycleGap = 50 * time.Second
```

Comment:
I don't think this works: an epoch in `fastnet` is only 1 minute. If the cyclegap is 50 seconds and the phaseshift 30. That would mean the poet round for the publish epoch n starts 30 seconds before epoch n starts and ends 10 seconds later - nodes will then already publish ATXs for epoch n in epoch n-1?

Reply:
We could maybe make Phaseshift 10 seconds and Cyclegap 40 to prevent this from happening?

Reply:
The node is waiting for the publish epoch to publish an ATX so it should work fine.

Reply:
I'm not sure about it, but multiple tests failed on `bors try` for the change - so I'm not sure if `Phaseshift + CycleGap` can be larger than `LayerDuration * LayersPerEpoch`

Reply:
That's not the issue.

- 12:00:00 - epoch n starts
- 12:00:30 - poet round for publish epoch n ends nodes generate PoST and publish ATX in epoch n start to wait until the start of the next poet round minus grace period
- 12:01:00 - epoch n+1 starts
- 12:01:10 - nodes register for poet round with publish epoch n+1
- 12:01:20 - poet round for epoch n+1 starts
- 12:01:30 - poet round for epoch n+1 ends, nodes fetch and start generating PoST proofs

the problem is at 12:01:10 - nodes are supposed to register for epoch n+1, but it is already epoch n+1 so I believe the nodes will discard the challenge as stale and skip the poet round...

Reply:
The code decides whether its "too late" by comparing the current time with the time when the poet round closes. At 12:00:10 it will see it still has 10s left.

Reply:
I checked the code and it might be OK that the publish epoch is the same as the current epoch when registering at PoET. However it seems like there are still some issues with system tests.